### ISSUE TITLE
`pnpm install` overrides `pnpm linked` modules

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#597]
### pnpm version:

0.51.1

### Code to reproduce the issue:

```
cd lodash
git clone https://github.com/lodash/lodash.git
pnpm link -g
cd ..
mkdir foo
cd foo
pnpm init -y
pnpm link lodash
pnpm installl
ls -al node_modules
```

### Expected behavior:

lodash should be linked to `.pnpm-global`.

### Actual behavior:

Its overidden.

### Additional information:

 - `node -v` prints:
 - Windows, OS X, or Linux?:

**Comment by vjpr** [comment]
Looks like it doesn't, perhaps I had a different version or something...

**Comment by vjpr** [comment]
It appears that it will keep the link if the linked module has the same version as in the `package.json`.

**Comment by zkochan** [comment]
lets reopen in that case

**Comment by vjpr** [comment]
A good approach might be to show a log message after a `pnpm i` to say which npm linked modules were changed and the potential issues.

E.g.

```
You have lodash symlinked to ~/dev/lodash.
You have react symlinked to ~/dev/react.
Redux was symlinked to ~/dev/redux, but the linked package.json version did not match this module's required version from the package.json. Therefore it was overridden.
```

**Comment by zkochan** [comment]
landed in [v0.55.3](https://github.com/pnpm/pnpm/releases/tag/v0.55.3)

**Comment by gaokun** [comment]
the same issue happened again on 7.x 
(haven't try 8.x yet)

@zkochan  Plz take a look, thx
